### DISCUSSION TITLE
Is it possible to use env variables into secrets' names?

### DISCUSSION THREAD

**Discussion by vitalyk-multinarity** [community/community#44458] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a few secrets, for example, KEY1_v2023, KEY2_v2023, KEY3_v2023.
Is there a more elegant way to use them instead of many lines like:
`echo ${{ secrets.KEY1_v2023 }} > .key1` 
?

I tried to generate a secret name from env variables like that (and use this in loop)
```
VERSION='v2023'
I="1"
SECRET_NAME="KEY${I}_${VERSION}"
echo  ${{ secrets.$SECRET_NAME }}
```

but it fails.

**Comment by jsoref** [comment, marked as answer]
it's important to understand the order of operations.

1. `${{ ... }}` evaluation
2. a script is written
3. your `shell` runs the script
4. your shell performs any of its own variable evaluation

---

* You could use a step before your step and have it generate outputs which would be available via `${{ ... }}`.
* You could write all your secrets to a file and then let your shell parse the file however you might choose to do so. (Obviously you should be careful about writing all your secrets to a file.)

**Reply by vitalyk-multinarity** [reply]
@jsoref - many thanks! I hope this time I'll remember this order.
